### ISSUE TITLE
JTReg Test Failure: java/lang/invoke/InvokeMethodHandleWithBadArgument.java

### ISSUE THREAD

**Issue by M-Davies** [eclipse/openj9#6943]
Failure link
------------
- https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/test/jdk/java/lang/invoke/InvokeMethodHandleWithBadArgument.java is consistantly failing for JDK13-openj9 (hotspot unaffected, fixed for JDK11-openj9)
- test category, openjdk
- OS/architecture, openjdk_x86-64_linux/osx, openjdk_x86-64_windows 
```
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+4-201909021901)
15:18:31  Eclipse OpenJ9 VM AdoptOpenJDK (build master-e2e21457f, JRE 11 Linux amd64-64-Bit Compressed References 20190902_324 (JIT enabled, AOT enabled)
15:18:31  OpenJ9   - e2e21457f
15:18:31  OMR      - 5628a974
15:18:31  JCL      - 259a016054 based on jdk-11.0.5+4)
```

Optional info
-------------

- intermittent failure (no)
- Both potentially caused by `asSpreader()` throwing the `IllegalArgumentException`. This throwable case could be added seeing as the test is supposed to catch this sort of exception anyway.

Failure output (captured from console output)
---------------------------------------------
```
15:22:14  STDOUT:
15:22:14  [TestNG] Running:
15:22:14    java/lang/invoke/InvokeMethodHandleWithBadArgument.java
15:22:14  
15:22:14  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithBadElementCount(): success
15:22:14  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithNull(): failure
15:22:14  java.lang.IllegalArgumentException
15:22:14  	at test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithNull(InvokeMethodHandleWithBadArgument.java:64)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
15:22:14  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
15:22:14  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
15:22:14  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
15:22:14  	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
15:22:14  	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
15:22:14  	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
15:22:14  	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
15:22:14  	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
15:22:14  	at org.testng.TestRunner.privateRun(TestRunner.java:773)
15:22:14  	at org.testng.TestRunner.run(TestRunner.java:623)
15:22:14  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
15:22:14  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
15:22:14  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
15:22:14  	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
15:22:14  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
15:22:14  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
15:22:14  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
15:22:14  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
15:22:14  	at org.testng.TestNG.run(TestNG.java:1018)
15:22:14  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
15:22:14  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:54)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
15:22:14  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
15:22:14  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
15:22:14  	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
15:22:14  	at java.base/java.lang.Thread.run(Thread.java:831)
15:22:14  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithZeroLength(): success
15:22:14  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithBadElementCount(): success
15:22:14  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithNull(): failure
15:22:14  java.lang.IllegalArgumentException
15:22:14  	at test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithNull(InvokeMethodHandleWithBadArgument.java:57)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
15:22:14  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
15:22:14  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
15:22:14  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
15:22:14  	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
15:22:14  	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
15:22:14  	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
15:22:14  	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
15:22:14  	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
15:22:14  	at org.testng.TestRunner.privateRun(TestRunner.java:773)
15:22:14  	at org.testng.TestRunner.run(TestRunner.java:623)
15:22:14  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
15:22:14  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
15:22:14  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
15:22:14  	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
15:22:14  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
15:22:14  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
15:22:14  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
15:22:14  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
15:22:14  	at org.testng.TestNG.run(TestNG.java:1018)
15:22:14  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
15:22:14  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:54)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
15:22:14  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
15:22:14  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
15:22:14  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
15:22:14  	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
15:22:14  	at java.base/java.lang.Thread.run(Thread.java:831)
15:22:14  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithZeroLength(): success
15:22:14  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testExactInvokerWithNull(): success
15:22:14  STDERR:
15:22:14  Unhandled exception
15:22:14  Type=Segmentation error vmState=0x00000000
15:22:14  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
15:22:14  Handler1=00007FAA89A42530 Handler2=00007FAA89327E80 InaccessibleAddress=000000000000000C
15:22:14  RDI=0000000001E8B300 RSI=0000000000000001 RAX=00007FAA84012B70 RBX=00007FAA4A5F3490
15:22:14  RCX=0000000000000008 RDX=0000000000000000 R8=000000000000003C R9=0000000000000004
15:22:14  R10=0000000000000000 R11=00007FAA8B057F90 R12=00000000E01212A8 R13=0000000001E92420
15:22:14  R14=00000000E012DB38 R15=0000000000000002
15:22:14  RIP=00007FAA89A15C73 GS=0000 FS=0000 RSP=00007FAA4A5F30E0
15:22:14  EFlags=0000000000010246 CS=E033 RBP=0000000000000000 ERR=0000000000000004
15:22:14  TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=000000000000000C
15:22:14  xmm0 0000000001e8b300 (f: 32027392.000000, d: 1.582363e-316)
15:22:14  xmm1 616c2f6176616a4c (f: 1986095744.000000, d: 1.981285e+161)
15:22:14  xmm2 00007faa4a5f3580 (f: 1247753600.000000, d: 6.935168e-310)
15:22:14  xmm3 00007faa4a5f3580 (f: 1247753600.000000, d: 6.935168e-310)
15:22:14  xmm4 00007faa4a5f3580 (f: 1247753600.000000, d: 6.935168e-310)
15:22:14  xmm5 6d75677241646142 (f: 1097097600.000000, d: 1.888920e+219)
15:22:14  xmm6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:22:14  xmm7 0000000000000002 (f: 2.000000, d: 9.881313e-324)
15:22:14  xmm8 3bbcc86800000000 (f: 0.000000, d: 6.095003e-21)
15:22:14  xmm9 3fbf6121fae9f556 (f: 4209636608.000000, d: 1.225759e-01)
15:22:14  xmm10 3d7544b7661012c2 (f: 1712329472.000000, d: 1.208970e-12)
15:22:14  xmm11 402ebe5e628de0eb (f: 1653465344.000000, d: 1.537181e+01)
15:22:14  xmm12 bcc5000000000000 (f: 0.000000, d: -5.828671e-16)
15:22:14  xmm13 bfbf6121fae9f580 (f: 4209636864.000000, d: -1.225759e-01)
15:22:14  xmm14 402e7f9c1e980d00 (f: 513281280.000000, d: 1.524924e+01)
15:22:14  xmm15 3c58c1dbef44561c (f: 4014233088.000000, d: 5.368375e-18)
15:22:14  Module=/home/jenkins/workspace/Grinder_Advanced/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9vm29.so
15:22:14  Module_base_address=00007FAA899A3000
15:22:14  Target=2_90_20190902_324 (Linux 4.4.0-141-generic)
15:22:14  CPU=amd64 (4 logical CPUs) (0x1f4f5a000 RAM)
15:22:14  ----------- Stack Backtrace -----------
15:22:14  (0x00007FAA89A15C73 [libj9vm29.so+0x72c73])
15:22:14  (0x00007FAA89A1D50B [libj9vm29.so+0x7a50b])
15:22:14  (0x00007FAA899BD236 [libj9vm29.so+0x1a236])
15:22:14  (0x00007FAA899BBE40 [libj9vm29.so+0x18e40])
15:22:14  (0x00007FAA89A7DBE2 [libj9vm29.so+0xdabe2])
15:22:14  ---------------------------------------
15:22:14  JVMDUMP039I Processing dump event "gpf", detail "" at 2019/09/03 14:22:09 - please wait.
15:22:14  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/core.20190903.142209.10858.0001.dmp' in response to an event
15:22:14  JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.10886.
15:22:14  
15:22:14  JVMDUMP010I System dump written to /home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/core.20190903.142209.10858.0001.dmp
15:22:14  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/javacore.20190903.142209.10858.0002.txt' in response to an event
15:22:14  JVMDUMP010I Java dump written to /home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/javacore.20190903.142209.10858.0002.txt
15:22:14  JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/Snap.20190903.142209.10858.0003.trc' in response to an event
15:22:14  JVMDUMP010I Snap dump written to /home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/Snap.20190903.142209.10858.0003.trc
15:22:14  JVMDUMP007I JVM Requesting JIT dump using '/home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/jitdump.20190903.142209.10858.0004.dmp'
15:22:14  JVMDUMP010I JIT dump written to /home/jenkins/workspace/Grinder_Advanced/openjdk-tests/TestConfig/test_output_15675205135959/jdk_custom_0/work/scratch/0/jitdump.20190903.142209.10858.0004.dmp
15:22:14  JVMDUMP013I Processed dump event "gpf", detail "".
```

**Comment by M-Davies** [comment]
Test passes for JDK11, reincluded. Now failing for JDK13 (error message is the same). @fengxue-IS would it be possible to reopen this issue and rebrand it as JDK13 only now?

**Comment by DanHeidinga** [comment]
@M-Davies  Can you provide links to the failing builds?

**Comment by M-Davies** [comment]
@DanHeidinga here is a recent one https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/783/

The link will likely expire soon so use this link if you need to rebuild the job https://ci.adoptopenjdk.net/job/Grinder/parambuild/?JDK_VERSION=13&JDK_IMPL=openj9&BUILD_LIST=openjdk&JenkinsFile=openjdk_x86-64_linux&CUSTOM_TARGET=test/jdk/java/lang/invoke/InvokeMethodHandleWithBadArgument.java&TARGET=jdk_custom_0

**Comment by DanHeidinga** [comment]
```
06:44:43  OpenJDK Runtime Environment AdoptOpenJDK (build 13+33)
06:44:43  Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.17.0-m2, JRE 13 Linux amd64-64-Bit Compressed References 20191009_90 (JIT enabled, AOT enabled)
06:44:43  OpenJ9   - 23b8b11b4
06:44:43  OMR      - 20db4fbc
06:44:43  JCL      - 2cb22d51d9 based on jdk-13+33)
```

The fix for this went in after the `m2` build occurred.  The fix is already in the 0.17 stream but not in this build

**Comment by M-Davies** [comment]
Thanks @DanHeidinga. That's fine then
